### PR TITLE
[Nim 2.0 Cleanup] 'lock levels' are deprecated, now a noop

### PR DESCRIPTION
**Pull request by ringabout**
remove lock levels logic while keeping locks sections

ref https://github.com/nim-lang/RFCs/issues/437

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -52,6 +52,8 @@ becomes an alias for `addr`.
 
 - Deprecated `selfExe` for Nimscript.
 
+- 'lock levels' is deprecated, now a noop.
```
Comment (konsumlamm) [inline changelog.md]:
```suggestion
- 'lock levels' are deprecated, now a noop.
```

File: compiler/pragmas.nim [hunk 2]
```diff
@@ -1189,7 +1173,7 @@ proc singlePragma(c: PContext, sym: PSym, n: PNode, i: var int,
       of wLocks:
         if sym == nil: pragmaLockStmt(c, it)
         elif sym.typ == nil: invalidPragma(c, it)
-        else: sym.typ.lockLevel = pragmaLocks(c, it)
+        else: warningDeprecated(c.config, n.info, "'lock levels' is deprecated, now a noop")
```
Comment (konsumlamm) [inline compiler/pragmas.nim]:
```suggestion
        else: warningDeprecated(c.config, n.info, "'lock levels' are deprecated, now a noop")
```

File: changelog.md [hunk 3]
```diff
@@ -62,7 +62,11 @@ becomes an alias for `addr`.
 - Removed deprecated `` httpcore.`==` ``.
 - Removed deprecated `std/dom_extensions`.
 - Removed deprecated `std/posix.CMSG_SPACE` and `std/posix.CMSG_LEN` that takes wrong argument types.
-- Removed deprecated `osproc.poDemon`, symbol with typo.
+
+- Deprecated `selfExe` for Nimscript.
+
+- Lock levels are deprecated, now a noop.
+
```
Comment (metagn) [inline changelog.md:66]: Bad merge?

File: changelog.md [hunk 4]
```diff
@@ -62,7 +62,11 @@ becomes an alias for `addr`.
 - Removed deprecated `` httpcore.`==` ``.
 - Removed deprecated `std/dom_extensions`.
 - Removed deprecated `std/posix.CMSG_SPACE` and `std/posix.CMSG_LEN` that takes wrong argument types.
-- Removed deprecated `osproc.poDemon`, symbol with typo.
+
+- Deprecated `selfExe` for Nimscript.
+
```
Comment (ringabout) [inline changelog.md]:
```suggestion
```

File: changelog.md [hunk 5]
```diff
@@ -62,7 +62,8 @@ becomes an alias for `addr`.
 - Removed deprecated `` httpcore.`==` ``.
 - Removed deprecated `std/dom_extensions`.
 - Removed deprecated `std/posix.CMSG_SPACE` and `std/posix.CMSG_LEN` that takes wrong argument types.
-- Removed deprecated `osproc.poDemon`, symbol with typo.
+- Lock levels are deprecated, now a noop.
```
Comment (metagn) [inline changelog.md:65]: This is still in standard library changes, also it removes a line

**Pull request closed without merging**